### ISSUE TITLE
Are there any tricks in this expression?

### ISSUE THREAD

**Issue by ThinkNaive** [fishmoon1234/DAG-GNN#12]
here's the line(52) in modules.py:
`logits = torch.matmul(adj_Aforz, x + self.Wa) - self.Wa`

whether is it same as the following line?
`logits = torch.matmul(adj_Aforz, x)`

I've test result between the two, however they are the same.
So I wonder if there were any tricks so that the param `Wa` is necessary?
Wish you could help me, thank you.

**Comment by ThinkNaive** [comment]
I found that expression in line(52) in modules.py `logits = torch.matmul(adj_Aforz, x + self.Wa) - self.Wa` is slightly different from that in the paper.

adj_Aforz denotes <a href="https://www.codecogs.com/eqnedit.php?latex=(I-A^T)" target="_blank"><img src="https://latex.codecogs.com/gif.latex?(I-A^T)" title="(I-A^T)" /></a>, use `A` for simplicity.
x denotes <a href="https://www.codecogs.com/eqnedit.php?latex={\rm{MLP}}(X,W^1,W^2)" target="_blank"><img src="https://latex.codecogs.com/gif.latex?{\rm{MLP}}(X,W^1,W^2)" title="{\rm{MLP}}(X,W^1,W^2)" /></a>, use `B` for simplicity.
As is mentioned in the paper, <a href="https://www.codecogs.com/eqnedit.php?latex=[M_Z|\log{S_Z}]=(I-A^T){\rm{MLP}}(X,W^1,W^2)" target="_blank"><img src="https://latex.codecogs.com/gif.latex?[M_Z|\log{S_Z}]=(I-A^T){\rm{MLP}}(X,W^1,W^2)" title="[M_Z|\log{S_Z}]=(I-A^T){\rm{MLP}}(X,W^1,W^2)" /></a>, so the corresponding expression is `logits = AB`.
However, expression in line(52) equals `AB+(∑∑Aij - 1)*Wa`

**Comment by helenlai** [comment]
same question, could you please explain what is the purpose of Wa here?
